commands/command_prune.go

```diff
@@ -56,6 +56,19 @@ type PruneProgress struct {
 }
 type PruneProgressChan chan PruneProgress
 
+func maxConcurrency() int64 {
+	fds, err := tools.GetMaxFileDescriptors()
+	if err != nil {
+		return int64(runtime.NumCPU() * 2)
+	}
+	// Assume 64 FDs for each operation and leave 64 for other purposes.
+	maxThreads := (int64(fds) - 64) / 64
+	if maxThreads < 1 {
+		return 1
+	}
+	return maxThreads
```

Comment:
Shouldn't this be `min(maxThreads, runtime.NumCPU() * 2)`?

Reply:
No, the old heuristic was to run twice as many goroutines as CPUs.  We can spawn arbitrarily many goroutines and the Go scheduler will only run as many as there are threads to run them (which is usually the number of CPUs).

The only limiting factor that's required here is file descriptors; the actual CPU usage is handled by the scheduler.  We retain the old check just to handle cases where we can't get a good value.